proto/sf/substreams/rpc/v2/service.proto

```diff
@@ -130,6 +130,37 @@ message OutputDebugInfo {
 
 message ModulesProgress {
   repeated ModuleProgress modules = 1;
```

Comment:
I think I would like to see some kind of "completeness" message where a percentage of completion is given. Once live (inside "live" segment), completion would be 100%. That is a key metric that will make implementation of sink must easier, for most cases, a simple `completion` like that reported each Xs is quite enough.

Reply:
hmmmm... could be complex, these metrics will serve "production" as well as "development" requests.
If you have a start-block at 15M, there is the notion of 'completeness' in back-processing before your start-block (why is it so slow before I get any data ?), then the notion of completeness for substreams on the whole chain... In this case, the "highest_processed_block" per module does give a sense of completeness, the definition of which can be (easily) computed on the client side.

I'm already thinking of cases where it will be difficult to determine "which" completeness means the most to the user (ex: development mode, or a substreams with only a mapper -- which can start at any block, production mode where the mapper cache-files have been produced only for certain specific ranges, ex: [10,000-20,000] but the dependency stores are ready up to block 15M...)

My approach here is to give correctly aggregated data, but I'm trying to avoid "computed subjective data", and I see 'completeness' as one of them...

You are welcome to change my mind :D